src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1473,6 +1473,20 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        private bool DetectRecursiveSymLink(DirectoryInfo path)
+        {
+            var readSymLink = Platform.NonWindowsInternalGetTarget(path.FullName);
+
+            if (path.FullName.Contains(readSymLink))
+            {
+                Console.WriteLine("\n Detected recursive symlink pointing to its parent directory.");
+                Console.WriteLine(" Skipping: " + path.FullName + " ==> " + readSymLink);
+                return true;
+            }
+
+            return false;
+        }
+
```

Comment:
This method seems only applicable to unix. In that case, you probably should guard it with `#if UNIX`.